jetty-io/src/main/java/org/eclipse/jetty/io/ByteBufferAccumulator.java

```diff
@@ -0,0 +1,115 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.io;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.jetty.util.BufferUtil;
+
+public class ByteBufferAccumulator implements AutoCloseable
+{
+    private static final int MIN_SPACE = 8;
+    private static final int DEFAULT_BUFFER_SIZE = 1024;
+
+    private final List<ByteBuffer> _buffers = new ArrayList<>();
+    private final ByteBufferPool _bufferPool;
+
+    public ByteBufferAccumulator()
+    {
+        this(null);
+    }
+
+    public ByteBufferAccumulator(ByteBufferPool bufferPool)
+    {
+        _bufferPool = (bufferPool == null) ? new NullByteBufferPool() : bufferPool;
+    }
+
+    public int getLength()
+    {
+        int length = 0;
+        for (ByteBuffer buffer : _buffers)
+            length += buffer.remaining();
+        return length;
+    }
+
+    public ByteBuffer ensureBuffer()
+    {
+        return ensureBuffer(DEFAULT_BUFFER_SIZE);
+    }
+
+    public ByteBuffer ensureBuffer(int minAllocationSize)
```

Comment:
So I was confused by the signature. So how about this:
```suggestion
    public ByteBuffer ensureBuffer(int minCapacity, int minAllicationSize)
```
So you pass in both the minim capacity and the size that you'd like to allocate at

Reply:
What is the minCapacity used for?

Reply:
It replaces your `MIN_SIZE` (which was wrongly named anyway)